File: Back/internal/models/project.go
package models

import "time"

type Project struct {
	Id            int
	Name          string
	Deadline      time.Time
	Pdf           string
	Image         string
	Collaborators []Collaborator
}

func NewProject() *Project {
	return &Project{
		Collaborators: make([]Collaborator, 0),
	}
}
func (p *Project) AddCollaborator(c Collaborator) {
	p.Collaborators = append(p.Collaborators, c)
}

File: Front/src/types/Project.ts
export interface project {
  Id: number;
  Name: string;
  Deadline: string;
  Pdf: string;
  Image: string;
  Collaborators: collaborator[];
}

export enum Tab {
  PDF = "Pdf",
  NOTES = "Notes",
  TASKS = "Tasks",
  GPT = "GPT",
}

export interface task{
  Id: number;
  projectId: number;
  Title: string;
  Description: string;
  // Deadline: {
  //   "Time": string;
  //   "valid": boolean;
  // }
  Completed: boolean;
}

export interface collaborator {
  Id: number;
  Name: string;
  Email: string;
  Tasks : task[];
}